koi/commands.py

```diff
@@ -99,21 +82,106 @@ def register_service(email, password, organisation_id, name=None,
     except socket.error as exc:
         raise click.ClickException(click.style(exc.strerror, fg='red'))
 
+    click.echo(click.style('\n{} service registered\n'.format(name),
+                           fg='green'))
+
+
+def _get_accounts_client(accounts_url, email, password):
+    """
+    Create an Accounts Service API client and log in using provided email and password
+    :param accounts_url: Accounts Service URL
+    :param email: Login Email
+    :param password: Login Password
+    :return: Accounts Service API Client
+    """
+    client = API(accounts_url, async=False, validate_cert=False)
+    response = client.accounts.login.post(email=email, password=password)
+    client.default_headers['Authorization'] = response['data']['token']
+    return client
+
+
+def _create_service(client, organisation_id, name, location, service_type):
+    """
+    Attempt to create service with given details. If service already exists look up existing service.
+    :param client: Accounts Service API Client
+    :param organisation_id: Id of Organisation
+    :param name: Service Name
+    :param location: Service Location
+    :param service_type: Service Type
+    :return: Service Id
+    """
     try:
-        with open(os.path.join(config, 'local.conf'), 'r') as f:
-            lines = [line for line in f.readlines()
-                     if not (line.startswith('service_id') or line.startswith('client_secret'))]
-    except IOError:
-        lines = []
+        response = client.accounts.organisations[organisation_id].services.post(
+            name=name, location=location, service_type=service_type)
+        service_id = response['data']['id']
+    except httpclient.HTTPError:
+        service_id = _get_service(client, organisation_id, name)
 
-    lines.append('\nservice_id = "{}"\n'.format(service_id))
+    return service_id
+
+
+def _get_service(client, organisation_id, name):
+    """
+    Get service belonging to organisation which matches given service name
+    :param client: Accounts Service API Client
+    :param organisation_id: Id of Organisation
+    :param name: Service Name
+    :return: Service Id
+    """
+    response = client.accounts.services.get(organisation_id=organisation_id)
+    services = [s for s in response['data'] if s['name'] == name]
+
+    if services:
+        return services[0]['id']
+    else:
+        msg = ('Organisation {} does not '
```

Comment:
If the organisation id does not exist it will raise a 404 error, which will be caught and reported by register service (same as the current workflow). Would it be better for it to do something else?